packages/number-field/src/NumberField.ts

```diff
@@ -495,7 +495,14 @@ export class NumberField extends TextfieldBase {
         // Step shouldn't validate when 0...
         if (this.step) {
             const min = typeof this.min !== 'undefined' ? this.min : 0;
-            const moduloStep = (value - min) % this.step;
+            this.digitsAfterDecimal =
```

Comment:
Can we cauterize this work to `change.get('step')` in one of the lifecycle callbacks? I'm not super excited about the `this._lastDigitsAfterDecimal !== this.digitsAfterDecimal` check, can we leverage the existing cache management mechanics?

Reply:
Sure!